src/api/app/services/notification_service/notifier.rb

```diff
@@ -0,0 +1,52 @@
+module NotificationService
+  class Notifier
+    EVENTS_TO_NOTIFY = ['Event::RequestStatechange',
+                        'Event::RequestCreate',
+                        'Event::ReviewWanted',
+                        'Event::CommentForProject',
+                        'Event::CommentForPackage',
+                        'Event::CommentForRequest'].freeze
+    CHANNELS = [:web, :rss].freeze
+
+    def initialize(event)
+      @event = event
+    end
+
+    def call
+      return unless @event.eventtype.in?(EVENTS_TO_NOTIFY)
+
+      CHANNELS.each do |channel|
+        @event.subscriptions(channel).each do |subscription|
+          create_notification_per_subscription(subscription, channel)
+        end
+      end
+    rescue StandardError => e
+      Airbrake.notify(e, event_id: @event.id)
+    end
+
+    private
+
+    def create_notification_per_subscription(subscription, channel)
+      return unless create_notification?(subscription.subscriber, channel)
+
+      klass = channel == :web ? 'Web' : 'RSS'
+      "NotificationService::#{klass}Channel".constantize.new(subscription, @event).call
```

Comment:
any other way to do this construction without the constantize?

Reply:
Can you explain why shouldn't I use the constantize?

Reply:
http://gavinmiller.io/2016/the-safesty-way-to-constantize/

Reply:
I see the problem.

I agree that using the `constantize` on data coming from the user is dangerous, but in this case the input is clearly limited to `Web` or `RSS`, so I see no problem here.

Anyway, I'm gonna rewrite it to avoid using the `constantize`

Reply:
Software engineering is programming over time. It's not so hard to write code that is correct today, the problem is the code calling this method tomorrow. Nobody review manually every called method to make sure that the parameter passed won't compromise it and to rewrite it safely using `constantize` is a PITA. Please avoid it.

Reply:
Changed, check it out again, please